### PR TITLE
Huobi futures api

### PR DESCRIPTION
**Pull request by lamtrhieu**
Hi @kimquy  @quantd2 
This is just work in progress but I would appreciate if you can spend some times review it first. 
I will try to add more tests tomorrow. 

Thanks

### CODE REVIEW

**Comment by kilu01** [comment]
I've merged this branch to my branch so anh @kimquy you just need to review https://github.com/acuityinnovations/huobi.ex/pull/2 which is the common structure that anh @lamtrhieu and me are agree upon

File: lib/ex_huobi/auth.ex [hunk 1]
```diff
@@ -0,0 +1,143 @@
+defmodule Huobi.Test do
```
Comment (sirius248) [inline lib/ex_huobi/auth.ex]: `Huobi.Test` should be `Huobi.Auth`

File: lib/ex_huobi/auth.ex [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  def timestamp() do
+    {timestamp, _} =
+      DateTime.utc_now() |> IO.inspect
+      |> DateTime.truncate(:second) |> IO.inspect
+      |> DateTime.to_iso8601() |> IO.inspect
+      |> String.split_at(-1) |> IO.inspect
```
Comment (sirius248) [inline lib/ex_huobi/auth.ex]: Pls remove `IO.inspect`

File: lib/ex_huobi/auth.ex [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    timestamp  
+  end
+
+  # def sign(method, host, path, params) do
+
+  def create() do
+    method = "POST\n"
```
Comment (sirius248) [inline lib/ex_huobi/auth.ex]: pls remove `\n`

File: lib/ex_huobi/auth.ex [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    signature =
+      :sha256
+      |> :crypto.hmac("94472b76-30a703ac-e75d7396-c34a0", presigned_text)
+      |> Base.encode64()
+
+    # api.huobi.pro
+    endpoint =
+      ("https://api.hbdm.com/api/v1/contract_order" <>
+         "?" <> params_string <> "&Signature=#{signature}")
+      |> IO.inspect()
```
Comment (sirius248) [inline lib/ex_huobi/auth.ex]: remove IO.inspect

File: lib/ex_huobi/auth.ex [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+      "Timestamp" => timestamp()
+    }
+
+    params_string = query_string("", params) |> IO.inspect()
+
+    presigned_text = method <> host <> path <> params_string
+
+    signature =
+      :sha256
+      |> :crypto.hmac("94472b76-30a703ac-e75d7396-c34a0", presigned_text)
```
Comment (sirius248) [inline lib/ex_huobi/auth.ex]: Not sure what is this `"94472b76-30a703ac-e75d7396-c34a0"`

File: lib/ex_huobi/rest/http_client.ex [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      |> DateTime.to_iso8601()
+      |> String.split_at(-1)
+
+    timestamp
+  end
+
+  def signed_path(method, host, path, params) do
+    %ExHuobi.Config{api_key: api_key, api_secret: api_secret} = ExHuobi.Config.get(nil)
+
+    api_key = "3c8dd81a-d8b5e97e-qv2d5ctgbn-9bd14"
```
Comment (sirius248) [inline lib/ex_huobi/rest/http_client.ex]: this should not be here

File: lib/ex_huobi/auth.ex (hunk 1 above)
Comment (lamtrhieu) [reply lib/ex_huobi/auth.ex]: @kimquy This file should be removed. I use it for testing. I will delete it now.

File: lib/ex_huobi/futures/rest/account.ex [hunk 7]
```diff
@@ -1,17 +1,20 @@
 defmodule ExHuobi.Futures.Rest.Account do
   alias ExHuobi.Rest.HTTPClient
+  alias ExHuobi.Futures.Util
 
-  @type params :: map
-  @type config :: ExHuobi.Config.t()
+  @type config :: ExHuobi.Config.config_or_nil
```
Comment (sirius248) [inline lib/ex_huobi/futures/rest/account.ex]: @lamtrhieu this is a bit unconventional.  Would suggest as following

```elixir
@type config :: ExHuobi.Config.t() | nil
```

File: lib/ex_huobi/futures/util.ex [hunk 8]
```diff
@@ -0,0 +1,27 @@
+defmodule ExHuobi.Futures.Util do
```
Comment (sirius248) [inline lib/ex_huobi/futures/util.ex]: I know naming is difficult. But in this case, `Util` not the best. can we rename to `ExHuobi.Futures.Handler`

File: lib/ex_huobi/margin/rest/account.ex [hunk 9]
```diff
@@ -4,15 +4,15 @@ defmodule ExHuobi.Margin.Rest.Account do
   @margin_endpoint "https://api.huobi.pro"
 
   @type params :: map
-  @type config :: Config.t()
+  @type config :: ExHuobi.Config.config_or_nil
```
Comment (sirius248) [inline lib/ex_huobi/margin/rest/account.ex]: Similar here `@type config :: ExHuobi.Config.t() | nil`

File: lib/ex_huobi/margin/rest/order.ex [hunk 10]
```diff
@@ -5,7 +5,7 @@ defmodule ExHuobi.Margin.Rest.Order do
 
   @type order_id :: integer
   @type params :: map | [map]
-  @type config :: Config.t()
+  @type config :: ExHuobi.Config.config_or_nil
```
Comment (sirius248) [inline lib/ex_huobi/margin/rest/order.ex]: `@type config :: ExHuobi.Config.t() | nil`

File: README.md [hunk 11]
```diff
@@ -64,3 +64,61 @@ ExHuobi.Margin.Rest.Order.bulk_create(
...
+    ]
+
+
+ExHuobi.Futures.Rest.Order.create_bulk_orders(orders, config)
+
+# Cancel order
+
+order_id = 1234567890
+
+ExHuobi.Futures.Rest.Order.create_bulk_orders(order_id, config)
```
Comment (sirius248) [inline README.md:122]: @lamtrhieu one last small comment. This is order bulk creation, why passing in only 1 single order id ?

Comment (lamtrhieu) [reply README.md:122]: @kimquy ah, it's a typo. Thanks for spot this out. Let me fix it.

**Review by kilu01 — APPROVED** [review]
LGTM

**Pull request merged**